### PR TITLE
[PROJECT] Module 18 - Context API & Hooks - StarWars Datatable

### PR DESCRIPTION
**Pull request by LeandroLM**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,193 @@
-# sd-01-week18-react-context-hooks-starwars-datatable-filters
+# Boas vindas ao repositório do projeto de Context API e Hooks!
+
+Você já usa o GitHub diariamente para desenvolver os exercícios, certo? Agora, para desenvolver os projetos, você deverá seguir as instruções a seguir. Fique atento a cada passo, e se tiver qualquer dúvida, nos envie por _Slack_! #vqv 🚀
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
Você já usa o GitHub diariamente para desenvolver os exercícios, certo? Agora, para desenvolver os projetos, você deverá seguir as instruções a seguir. Fique atento a cada passo e, se tiver qualquer dúvida, nos envie por _Slack_! #vqv 🚀
```

File: README.md [hunk 2]
```diff
@@ -1 +1,193 @@
...
+
+Caso não tenha realizado o projeto anterior, você deverá implementar todos os requisitos obrigatórios do projeto anterior e os novos requisitos do projeto atual. Contudo, não é necessário primeiro desenvolver o projeto com Redux para então substituí-lo por Context API e Hooks.
+
+Um dos requisitos deste projeto é a adição de testes. Recomendamos que você escreva os testes e garanta uma boa cobertura antes de começar a trabalhar na refatoração. Dessa forma, durante o desenvolvimento, você terá segurança de que sua aplicação continua funcionando conforme especificado, sem precisar testá-lo manualmente a todo momento. Além disso, você treinará escrever testes baseados em comportamento - se seus testes começaram a falhar após a transição, é um sinal de que eram dependentes de implementação e não do comportamento da sua aplicação. Este processo - escrever agnósticos em relação a implementação - não é fácil, mas é uma habilidade importante de se desenvolver.
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
Um dos requisitos deste projeto é a adição de testes. Recomendamos que você escreva os testes e garanta uma boa cobertura antes de começar a trabalhar na refatoração. Dessa forma, durante o desenvolvimento, você terá segurança de que sua aplicação continua funcionando conforme especificado, sem precisar testá-lo manualmente a todo momento. Além disso, você treinará escrever testes baseados em comportamento - se seus testes começaram a falhar após a transição, é um sinal de que eram dependentes de implementação e não do comportamento da sua aplicação. Este processo - escrever agnósticos em relação à implementação - não é fácil, mas é uma habilidade importante de se desenvolver.
```

File: README.md [hunk 3]
```diff
@@ -1 +1,193 @@
...
+```shell
+cd <diretorio onde estão seus projetos>
+cp -av sd-01-week16-react-redux-starwars-datatable-filters/src/ sd-01-week18-react-context-hooks-starwars-datatable-filters/src/
+```
+
+Atente para a barra (`/`) ao final dos nomes dos diretórios. Adapte os comandos com os nomes dos diretórios onde estão seus repositórios em sua máquina local.
+
+## Requisitos do projeto
+
+A seguir estão listados todos os requisitos do projeto. Para conveniência, os requisitos do projeto 16 foram transcritos aqui, com as devidas adaptações, onde necessário, para o contexto do atual projeto.
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
A seguir, estão listados todos os requisitos do projeto. Para conveniência, os requisitos do projeto 16 foram transcritos aqui, com as devidas adaptações, quando necessário, para o contexto do atual projeto.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,193 @@
...
+
+### 1 - A cobertura de testes do projeto deve ser maior que ou igual a 90%
+
+### 2 - O projeto deve ser feito, necessariamente, utilizando Context API e Hooks para controlar estado
+
+Não pode haver nenhum nenhum tipo de código Redux no seu projeto - _stores_, _actions_, _reducers_, _action creators_. Lembre-se de desinstalar o **Redux** e quais dependências relacionadas após a conclusão, caso seu projeto dê continuidade ao projeto anterior.
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
Não pode haver nenhum nenhum tipo de código Redux no seu projeto - _stores_, _actions_, _reducers_, _action creators_. Lembre-se de desinstalar o **Redux** e quaisquer dependências relacionadas após a conclusão, caso seu projeto dê continuidade ao projeto anterior.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,193 @@
...
+
+O estado especificado nos requisitos deve ser controlado fazendo uso **unicamente** de **Context API** e **Hooks**. 
+
+Contudo, caso julgue necessário manter informações de estado além do especificado nestes requisitos, você poderá decidir entre armazenar tal estado em um componente ou em um contexto, mas não no **Redux**.
+
+### 3 - Todos os componentes da aplicação devem ser componentes de função
+
+O uso de **Hooks** torna desnecessário a criação de componentes de classe. Portanto, você deverá converter quaiser componentes de classe que tenha escrito para componentes de função.
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
O uso de **Hooks** torna desnecessária a criação de componentes de classe. Portanto, você deverá converter quaisquer componentes de classe que tenha escrito para componentes de função.
```

File: README.md [hunk 6]
```diff
@@ -1 +1,193 @@
...
+
+2. Instale as dependências
+  * `npm install`
+
+3. Crie uma branch a partir da branch `master`
+  * Verifique que você está na branch `master`
+    * Exemplo: `git branch`
+  * Se não estiver, mude para a branch `master`
+    * Exemplo: `git checkout master`
+  * Agora crie uma branch para qual você vai submeter os `commits` do seu projeto
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
  * Agora crie uma branch para a qual você vai submeter os `commits` do seu projeto
```

File: README.md [hunk 7]
```diff
@@ -1 +1,193 @@
...
+
+Se ainda houver alguma dúvida sobre como entregar seu projeto, [aqui tem um video explicativo](https://vimeo.com/362189205).
+
+---
+
+### REVISANDO UM PULL REQUEST
+
+⚠⚠⚠
+
+À medida que você e os outros alunos forem entregando os projetos, vocês serão alertados **via Slack** para também fazer a revisão dos _Pull Requests_ dos seus colegas. Fiquem atentos às mensagens do _"Pull Reminders"_ no _Slack_!
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
À medida que você e os outros estudantes forem entregando os projetos, vocês serão alertados **via Slack** para também fazer a revisão dos _Pull Requests_ dos seus colegas. Fiquem atentos às mensagens do _"Pull Reminders"_ no _Slack_!
```

File: README.md [hunk 8]
```diff
@@ -1 +1,193 @@
...
+
+Os monitores também farão a revisão de todos os projetos, e irão avaliar tanto o seu _Pull Request_, quanto as revisões que você fizer nos _Pull Requests_ dos seus colegas!!!
```
Comment (betinaribeiro) [inline README.md]:
```suggestion
Os monitores também farão a revisão de todos os projetos, e vão avaliar tanto o seu _Pull Request_, quanto as revisões que você fizer nos _Pull Requests_ dos seus colegas!!!
```

File: README.md [hunk 9]
```diff
@@ -1 +1,193 @@
...
+
+Você já usa o GitHub diariamente para desenvolver os exercícios, certo? Agora, para desenvolver os projetos, você deverá seguir as instruções a seguir. Fique atento a cada passo e, se tiver qualquer dúvida, nos envie por _Slack_! #vqv 🚀
+
+Aqui você vai encontrar os detalhes de como estruturar o desenvolvimento do seu projeto a partir deste repositório, utilizando uma branch específica e um _Pull Request_ para colocar seus códigos.
+
+## O que deverá ser desenvolvido
+
+Você vai refatorar um projeto em React desenvolvido no módulo 16, [Starwars Datatable Filters](https://github.com/tryber/sd-01-week18-react-context-hooks-starwars-datatable-filters), substituindo React por Context API e Hooks para controlar estado.
+
+Você também adicionará testes a sua aplicação antes de refatorá-la, de forma a garantir o mesmo comportamento após a transição de Redux para Context API e Hooks.
```
Comment (albertosca) [inline README.md]:
```suggestion
Você também adicionará testes à sua aplicação antes de refatorá-la, de forma a garantir o mesmo comportamento após a transição de Redux para Context API e Hooks.
```

File: README.md [hunk 10]
```diff
@@ -1 +1,193 @@
...
+
+## Desenvolvimento
+
+Este repositório já contém um _template_ de uma aplicação React criado e configurado. Após clonar o projeto e instalar as dependências (mais sobre isso abaixo), você deverá copiar os arquivos que criou ou modificou no projeto anterior para este repositório. Veja [aqui](#copiando-os-arquivos-entre-projetos) instruções de como fazer isso de uma forma fácil.
+
+Caso não tenha realizado o projeto anterior, você deverá implementar todos os requisitos obrigatórios do projeto anterior e os novos requisitos do projeto atual. Contudo, não é necessário primeiro desenvolver o projeto com Redux para então substituí-lo por Context API e Hooks.
```
Comment (albertosca) [inline README.md]: Good one! This way we partially avoid the potential issues caused by the project's dependency on the other one.

File: README.md [hunk 11]
```diff
@@ -1 +1,193 @@
...
+
+Um dos requisitos deste projeto é a adição de testes. Recomendamos que você escreva os testes e garanta uma boa cobertura antes de começar a trabalhar na refatoração. Dessa forma, durante o desenvolvimento, você terá segurança de que sua aplicação continua funcionando conforme especificado, sem precisar testá-lo manualmente a todo momento. Além disso, você treinará escrever testes baseados em comportamento - se seus testes começaram a falhar após a transição, é um sinal de que eram dependentes de implementação e não do comportamento da sua aplicação. Este processo - escrever agnósticos em relação à implementação - não é fácil, mas é uma habilidade importante de se desenvolver.
```
Comment (albertosca) [inline README.md]:
```suggestion
Um dos requisitos deste projeto é a adição de testes. Recomendamos que você escreva os testes e garanta uma boa cobertura antes de começar a trabalhar na refatoração. Dessa forma, durante o desenvolvimento, você terá segurança de que sua aplicação continua funcionando conforme especificado, sem precisar testá-la manualmente a todo momento. Além disso, você treinará escrever testes baseados em comportamento - se seus testes começaram a falhar após a transição, é um sinal de que eram dependentes de implementação e não do comportamento da sua aplicação. Este processo - escrever testes agnósticos em relação à implementação - não é fácil, mas é uma habilidade importante de se desenvolver.
```

File: README.md [hunk 12]
```diff
@@ -1 +1,193 @@
...
+
+## Requisitos do projeto
+
+A seguir, estão listados todos os requisitos do projeto. Para conveniência, os requisitos do projeto 16 foram transcritos aqui, com as devidas adaptações, quando necessário, para o contexto do atual projeto.
+
+### 1 - A cobertura de testes do projeto deve ser maior que ou igual a 90%
+
+### 2 - O projeto deve ser feito, necessariamente, utilizando Context API e Hooks para controlar estado
+
+Não pode haver nenhum nenhum tipo de código Redux no seu projeto - _stores_, _actions_, _reducers_, _action creators_. Lembre-se de desinstalar o **Redux** e quaisquer dependências relacionadas após a conclusão, caso seu projeto dê continuidade ao projeto anterior.
```
Comment (jeanpsv) [inline README.md]:
```suggestion
Não pode haver nenhum tipo de código Redux no seu projeto - _stores_, _actions_, _reducers_, _action creators_. Lembre-se de desinstalar o **Redux** e quaisquer dependências relacionadas após a conclusão, caso seu projeto dê continuidade ao projeto anterior.
```

File: README.md [hunk 13]
```diff
@@ -1 +1,193 @@
...
+
+### 8 - Cada filtro de valores numéricos deve ter um ícone de `X` que, ao ser clicado, o apaga e desfaz suas filtragens dos dados da tabela
+
+A coluna que este filtro selecionava deve passar a ficar disponível nos dropdowns dos demais filtros já presentes na tela. Você deve usar **Context API** e **Hooks** para fazer o gerenciamento do estado da aplicação.
+
+## BÔNUS
+
+### 9 - As colunas da tabela devem ser ordenáveis de forma ascendente ou descendente
+
+A informação acerca da ordenação das colunas deve ser armazenada nos campos `filters: { column: 'Name', order: 'ASC'}`, o campo column representa o nome da coluna a ordenar e a ordem representa a ordenação, sendo 'ASC' ascendente e 'DESC' descendente. Por padrão, a tabela começa ordenada pela coluna 'Name' em ordem ascendente.
```
Comment (jeanpsv) [inline README.md]:
```suggestion
A informação acerca da ordenação das colunas deve ser armazenada nos campos `filters: { column: 'Name', order: 'ASC'}`, o campo column representa o nome da coluna a ordenar e order representa a ordenação, sendo 'ASC' ascendente e 'DESC' descendente. Por padrão, a tabela começa ordenada pela coluna 'Name' em ordem ascendente.
```

File: README.md [hunk 14]
```diff
@@ -1 +1,193 @@
...
+
+---
+
+### REVISANDO UM PULL REQUEST
+
+⚠⚠⚠
+
+À medida que você e os outros estudantes forem entregando os projetos, vocês serão alertados **via Slack** para também fazer a revisão dos _Pull Requests_ dos seus colegas. Fiquem atentos às mensagens do _"Pull Reminders"_ no _Slack_!
+
+Os monitores também farão a revisão de todos os projetos, e vão avaliar tanto o seu _Pull Request_, quanto as revisões que você fizer nos _Pull Requests_ dos seus colegas!!!
```
Comment (jeanpsv) [inline README.md]: We do not review the code. I think this part should be removed. What do you think?

File: README.md (hunk 10 above)
Comment (LeandroLM) [reply README.md]: https://mundoeducacao.bol.uol.com.br/gramatica/crase-antes-pronomes-possessivos.htm 😬

File: README.md (hunk 9 above)
Comment (LeandroLM) [reply README.md]: https://mundoeducacao.bol.uol.com.br/gramatica/crase-antes-pronomes-possessivos.htm 😬

Comment (albertosca) [reply README.md]: 🐺 
Let it be registered that the use is _optional_, not _wrong_ 😂

**Pull request merged**